src/modules/Categories.js

```diff
@@ -0,0 +1,227 @@
+          reject(err);
+        });
+    });
+  }
+
+  getComments = async (id) => {
+    const url = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${this.myId}/comments?item_id=${id}`;
+    const result = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    return result;
+  };
+
+  initApp = async () => {
+    const url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/';
+    const result = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    result.then((res) => { res.text().then((a) => console.log(a)); });
+  };
+
+  addComment = async (itemId, name, comment) => {
+    const url = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${this.myId}/comments`;
+    console.log(url);
+    const result = fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(
+        {
+          item_id: itemId,
+          username: name,
+          comment,
+        },
+      ),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+
+    const promise1 = result.then((res) => { res.text().then((a) => console.log(a)); });
+    await promise1;
+  }
+
+  printComments = (comments) => {
+    comments.then((res) => {
+      res.json().then((data) => {
+        const commentsTitle = this.popupComments.querySelector('.comments-title');
+        commentsTitle.innerHTML = `Commments (${data.length})`;
+        const commentsContainer = this.popupComments.querySelector('.comments-container');
+        commentsContainer.innerHTML = '';
+        let newData;
+        for (let i = 0; i < data.length; i += 1) {
+          newData = document.createElement('li');
+          newData.innerHTML = `${data[i].creation_date} ${data[i].username}: ${data[i].comment}`;
+          commentsContainer.appendChild(newData);
+        }
+      });
+    });
+  }
+
+  countComments = (comments) => {
+    comments.then((res) => {
+      res.json().then((data) => {
+        const commentsTitle = this.popupComments.querySelector('.comments-title');
+        commentsTitle.innerHTML = `Commments (${data.length})`;
+        const commentsContainer = this.popupComments.querySelector('.comments-container');
+        commentsContainer.innerHTML = '';
+        let newData;
+        for (let i = 0; i < data.length; i += 1) {
+          newData = document.createElement('li');
+          newData.innerHTML = `${data[i].creation_date} ${data[i].username}: ${data[i].comment}`;
+          commentsContainer.appendChild(newData);
+        }
+      });
+    });
+  }
+
+  getItemApiMain = async (mealId) => {
+    const url = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${mealId}`;
+    const result = fetch(url);
+
+    await result.then((response) => response.json()).then((json) => {
```

Comment:
the response.json() method returns a promise that when resolved returns the object parsed from JSON syntax, it does NOT return JSON. So instead of naming the variable JSON, consider changing its name for something more understandable, like mealDetails or something like that.